Remotion/Validation/UnitTests/Rules/PropertyValidationRuleTest.cs

```diff
@@ -1,33 +1,250 @@
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using System;
+using Moq;
+using NUnit.Framework;
+using Remotion.Reflection;
+using Remotion.Validation.Results;
 using Remotion.Validation.Rules;
 using Remotion.Validation.UnitTests.TestDomain;
+using Remotion.Validation.Validators;
 
 namespace Remotion.Validation.UnitTests.Rules
 {
   [TestFixture]
   public class PropertyValidationRuleTest
   {
     [Test]
-    [Ignore("TODO RM-5906")]
-    public void Validate ()
+    public void Validate_WithNullAsInstanceToValidate_SkipsValidation ()
     {
+      var context = new ValidationContext(null);
+
+      var validatorMock = new Mock<IPropertyValidator>();
+      validatorMock
+          .Setup(_ => _.Validate(It.IsAny<PropertyValidatorContext>()))
+          .Throws(new AssertionException("Should not be called."));
+
+      var propertyInformationMock = new Mock<IPropertyInformation>();
+
+      Func<object, object> propertyFunc = obj => null;
+
+      var rule = new PropertyValidationRule<object, int>(propertyInformationMock.Object, propertyFunc, null, new[] { validatorMock.Object });
+
+      var result = rule.Validate(context);
+
+      Assert.That(result, Is.Empty);
+    }
+
+    [Test]
+    public void Validate_WithConditionIsFalse_SkipsValidation ()
+    {
+      var context = new ValidationContext(new Person());
+
+      var validatorMock = new Mock<IPropertyValidator>();
+
+      validatorMock
+          .Setup(_ => _.Validate(It.IsAny<PropertyValidatorContext>()))
+          .Throws(new AssertionException("Should not be called."));
+
+      var propertyInformationMock = new Mock<IPropertyInformation>();
+
+      Func<object, object> propertyFunc = obj => null;
+      Func<object, bool> conditionFunc = _ => false;
+
+      var rule = new PropertyValidationRule<object, int>(propertyInformationMock.Object, propertyFunc, conditionFunc, new[] { validatorMock.Object });
+
+      var result = rule.Validate(context);
+
+      Assert.That(result, Is.Empty);
+    }
+
+    [Test]
+    public void Validate_WithConditionIsTrue_ReturnsValidationFailures ()
+    {
+      var context = new ValidationContext(new Person());
+      var propertyInformationMock = new Mock<IPropertyInformation>();
+
+      var propertyFailure = new PropertyValidationFailure(new Person(), propertyInformationMock.Object, "propval", "err", "validmessage");
+
+      var validatorMock = new Mock<IPropertyValidator>();
+
+      PropertyValidatorContext propertyValidatorContext = null;
+      validatorMock
+          .Setup(_ => _.Validate(It.IsAny<PropertyValidatorContext>()))
+          .Callback((PropertyValidatorContext ctx) => { propertyValidatorContext = ctx; })
+          .Returns(new[] { propertyFailure });
+
+
+      object propertyObj = null;
+      object propertyValueObj = null;
+      Func<object, object> propertyFunc = obj =>
+      {
+        propertyObj = obj;
+        propertyValueObj = "test";
+        return propertyValueObj;
```

Comment:
You could just return "test" here, since `propertyValueObj` seems to not have any other uses.